lldb/test/CMakeLists.txt

```diff
@@ -29,6 +29,22 @@ if(LLDB_ENFORCE_STRICT_TEST_REQUIREMENTS)
         "`LLDB_ENFORCE_STRICT_TEST_REQUIREMENTS=OFF`")
     endif()
   endforeach()
+
+  # On Windows make is not part of the MSYS tools that llvm-lit takes care of
+  if(LLDB_TEST_MAKE)
+    set(LLDB_DEFAULT_TEST_MAKE ${LLDB_TEST_MAKE})
+  else()
+    find_program(LLDB_DEFAULT_TEST_MAKE make)
+    if(LLDB_DEFAULT_TEST_MAKE)
+      message(STATUS "Found make: ${LLDB_DEFAULT_TEST_MAKE}")
```

Comment:
Is there a reason to conditionalize this on `LLDB_ENFORCE_STRICT_TEST_REQUIREMENTS`? I would hoist this up so that we always have CMake configure `make` and only conditionalize the warning.

Reply:
Well, no there is no GOOD reason. For the moment it was easy to keep it guarded by `LLDB_ENFORCE_STRICT_TEST_REQUIREMENTS` since it is `OFF` by default and won't fall on people's feet immediately. However, I am afraid we should special-case NetBSD and FreeBSD here to look for `gmake` instead of `make`.

Let me fix this in a follow-up PR that includes the author of https://github.com/llvm/llvm-project/pull/93883

Reply:
How about doing it in the other order? I think it would be somewhat strange if `LLDB_ENFORCE_STRICT_TEST_REQUIREMENTS` actually changed what is being used (instead of just erroring out when detecting errors).

That said, I think that:
- this PR is small enough to do both of those things together
- all that's needed is to do something like `find_program(LLDB_DEFAULT_TEST_MAKE NAMES gmake make)`. `gmake` can exist on non-netbsd systems as well, and it should not be an error to use that (in fact, `make` is a symlink to `gmake` on my system)

Reply:
@labath Thanks for your feedback! I already set up the follow-up. I will take your proposal into account. Hope it's ok to do it there: https://github.com/llvm/llvm-project/pull/111744

Reply:
Yes, in the end it's small enough indeed. Let me abandon https://github.com/llvm/llvm-project/pull/111744 and add these changes here.